File: 976-largest-perimeter-triangle/976-largest-perimeter-triangle.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort(reverse = True)
        
        for i in range(len(nums) - 2):
            if nums[i + 1] + nums[i + 2] > nums[i]:
                return nums[i] + nums[i + 1] + nums[i + 2]
        
        return 0
        

File: 976-largest-perimeter-triangle/976-largest-perimeter-triangle.cpp
class Solution {
public:
    int largestPerimeter(vector<int>& nums) {
        sort(nums.begin(), nums.end(), greater<>());
        int n = nums.size();
        
        for(int i = 0; i < n - 2; ++i) {
            if(nums[i] < nums[i + 1] + nums[i + 2])
                return nums[i] + nums[i + 1] + nums[i + 2];
        } 
        return 0;
    }
};